### PR TITLE
7874: Remove need for helpers in base/containers/utils.h and drop file

### PR DESCRIPTION
**Pull request by mariospr**
Resolves https://github.com/brave/brave-browser/issues/7874

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
N/A

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/brave-ios/Shared/BATCommonOperations.mm [hunk 1]
```diff
@@ -102,16 +102,16 @@ - (void)loadURLRequest:(const std::string &)url headers:(const std::vector<std::
     if (error) {
       errorDescription = error.localizedDescription.UTF8String;
     }
-    // For some reason I couldn't just do `std::map<std::string, std::string> responseHeaders;` due to std::map's
-    // non-const key insertion
-    auto* responseHeaders = new std::map<std::string, std::string>();
+    // For some reason I couldn't just do `base::flat_map<std::string, std::string>
+    // responseHeaders;` due to base::flat_map's non-const key insertion
+    auto* responseHeaders = new base::flat_map<std::string, std::string>();
```
Comment (kylehickinson) [inline vendor/brave-ios/Shared/BATCommonOperations.mm:107]: Does this compile? I assume based on my original comment, I tried to use `base::flat_map` but couldn't

Comment (mariospr) [reply vendor/brave-ios/Shared/BATCommonOperations.mm:107]: I don't know yet, to be honest. I don't have a Mac around so, I focused mainly on Linux and Android so far and so I made this change in the .mm files as carefully as I could by reading the code and changing what I think needed changing (i.e. the `loadURLRequest()` method from `BATCommonOperations.{h,mm}`), and I was now hoping for the bots to tell me whether it builds or not.

In the other hand, I don't understand well what you mean by "I assume based on my original comment", I don't see a reference to `base::flat_map` at all there, just to `std::map`. Could you please clarify? Thanks!

Comment (kylehickinson) [reply vendor/brave-ios/Shared/BATCommonOperations.mm:107]: Your right, sorry, the find + replace for all the std::map's in that comment got me confused 😅

**Comment by mariospr** [comment]
It looks like it has finally built fine in all platforms 👌

@reviewers PTAL when you have a moment, thanks!

**Review by tmancey — APPROVED** [review]
LGTM++

**Review by kylehickinson — APPROVED** [review]
iOS changes lgtm

**Review by mkarolin — APPROVED** [review]
++

**Review by zenparsing — APPROVED** [review]

**Comment by mariospr** [comment]
Thank you all for the reviews, just force-pushed now to resolve conflicts and will be merging once it's confirmed in the bots.

**Comment by mkarolin** [comment]
Looks like `patches/chrome-VERSION.patch` snuck in. It shouldn't be included.

**Comment by mariospr** [comment]
> Looks like patches/chrome-VERSION.patch snuck in. It shouldn't be included.

Removed that change and merged to master

**Pull request merged**